src/http/Request.ts

```
@@ -19,6 +20,29 @@ export class Request extends RequestProperties {
     constructor(httpInput: RequestProperties) {
         super();
         Object.assign(this, httpInput);
+        if (httpInput.headers['x-ms-client-principal']) {

```

If there's any issues parsing the client principal, I don't think that should throw an error here (basically preventing the entire execution from happening). Instead, how about we lazy-parse this by putting it in a getter - that way only people that try to access the user property would see the error.

Only downside to a getter is by default it would re-do the parsing each time. However, we can get around that pretty easily by just caching the result of our parse. Something like this:
```typescript
private _cachedUser?: HttpRequestUser;
public get user(): HttpRequestUser | null {
    if (this._cachedUser === undefined) {
        this._cachedUser = // do all your stuff to parse user
    }

    return this._cachedUser;
}
```